File: app/src/main/java/com/github/overpass/gather/auth/register/confirm/ConfirmEmailUseCase.java
package com.github.overpass.gather.auth.register.confirm;

import com.github.overpass.gather.SingleLiveEvent;

public class ConfirmEmailUseCase {

    private final ConfirmEmailRepo confirmEmailRepo;

    public ConfirmEmailUseCase(ConfirmEmailRepo confirmEmailRepo) {
        this.confirmEmailRepo = confirmEmailRepo;
    }

    public void confirmEmail(SingleLiveEvent<ConfirmEmailStatus> confirmEmailData) {
        confirmEmailRepo.confirmEmail(confirmEmailData);
    }
}

File: app/src/main/java/by/overpass/gather/model/usecase/confirm/ConfirmEmailUseCase.kt
package by.overpass.gather.model.usecase.confirm

import by.overpass.gather.data.repo.confirm.ConfirmEmailRepo
import com.google.android.gms.tasks.Task
import javax.inject.Inject

class ConfirmEmailUseCase @Inject constructor(private val confirmEmailRepo: ConfirmEmailRepo) {

    fun confirmEmail(): Task<Void> = confirmEmailRepo.confirmEmail()

    fun isEmailVerified(): Boolean = confirmEmailRepo.isEmailVerified()
}
